### PR TITLE
Fix bug 1528178 - Format Fluent strings in string list and tools as simple strings

### PR DESCRIPTION
**Pull request by adngdb**
@mathjazz Opening this early because I'm going full TDD here. Can you take a quick look at the tests and tell me if they are correct? 

Note to anyone reading this: if you haven't seen *Avengers: Endgame* and you care about it, don't read this pull request. :-)

### CODE REVIEW

**Comment by mathjazz** [comment]
Well done!

I would define `message` and `res` in also in the first two cases for consistency.

**Comment by adngdb** [comment]
@mathjazz Well, that turned out to be a loooot simpler than I feared, thanks to your and @Pike's code. So thanks, and yay Fluent preview!

**Comment by adngdb** [comment]
> I would define `message` and `res` in also in the first two cases for consistency.

I actually like to show that these tests are different from the rest, and do not provide an action Fluent message but a junk value.

File: frontend/src/core/utils/fluent/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* @flow */
+
+import getSimplePreview from './getSimplePreview';
+
+
+export default {
+    getSimplePreview,
+};
```
[the remark this answers is not in the document]
Comment (adngdb) [reply frontend/src/core/utils/fluent/index.js]: Yeah, I'd rather do that when it makes sense. Right now it doesn't for just one function.

File: frontend/src/modules/history/components/Translation.js [hunk 2]
```diff
@@ -316,11 +322,11 @@ export class TranslationBase extends React.Component<InternalProps, State> {
                         <TranslationPlaceablesDiff
                             diffTarget={ activeTranslation.string }
```
Comment (mathjazz) [inline frontend/src/modules/history/components/Translation.js]: This one needs to be optimized as well.

File: frontend/src/core/placeable/components/WithPlaceablesForFluent.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    // Insert after the last space-related rule.
+    let insertAfter = newRules.indexOf(multipleSpaces);
+    newRules.splice(insertAfter, 0, fluentFunction);
+    newRules.splice(insertAfter++, 0, fluentString);
+    newRules.splice(insertAfter++, 0, fluentParametrizedTerm);
+    newRules.splice(insertAfter++, 0, fluentTerm);
+
+    return newRules;
+}
```
[the remark this answers is not in the document]
Comment (adngdb) [reply frontend/src/core/placeable/components/WithPlaceablesForFluent.js]: That would collide with Python format strings I think.

**Comment by adngdb** [comment]
@mathjazz Hopefully this is complete now!

File: frontend/src/core/placeable/components/WithPlaceablesNoLeadingSpace.js [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
 import { rules } from './WithPlaceables';
 import leadingSpace from '../parsers/leadingSpace';
+import unusualSpace from '../parsers/unusualSpace';
 
 
-function getRulesWithoutLeadingSpace(rules: Array<Object>) {
+export function getRulesWithoutLeadingSpace(rules: Array<Object>) {
     let newRules = [ ...rules ];
     newRules.splice(newRules.indexOf(leadingSpace), 1);
+    newRules.splice(newRules.indexOf(unusualSpace), 1);
```
Comment (mathjazz) [inline frontend/src/core/placeable/components/WithPlaceablesNoLeadingSpace.js]: Thank you!

File: frontend/src/core/placeable/components/WithPlaceablesForFluentNoLeadingSpace.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Component that marks placeables in a string. Same as WithPlaceables but
+ * without the leadingSpace parser.
+ *
+ * The leadingSpace parser checks for spaces at the beginning of a line.
+ * If the input was previously split, then that parser will generate a lot of
+ * false positive (for example, if something else was marked just before a
+ * space, it thus becomes a space the beginning of a line). We thus want to
+ * have a special Placeables component without that parser, for use in
+ * combination with other parsing tools (like diff).
```
Comment (adngdb) [inline frontend/src/core/placeable/components/WithPlaceablesForFluentNoLeadingSpace.js]: Forgot to update this comment.

**Pull request merged**